### DISCUSSION TITLE
Does connection pooling works similar in valkey-go like it works in go-redis

### DISCUSSION THREAD

**Discussion by royatanu94** [valkey-io/valkey#793] [Q&A]
Does connection pooling works similar in valkey-go like it works in go-redis ?
Please share some docs for it

**Comment by rueian** [comment]
No, it doesn't by default. Valkey-go uses long-live connections to enable auto pipelining and client-side caching. That is, it doesn't occupy a connection for each request but it pipelines requests to a shared connection continuously.
